Keda.Cosmosdb.Scaler/src/Services/CosmosDBExternalScaler.cs

```diff
@@ -0,0 +1,173 @@
+                trigger.AccountName = accountName;
+            }
+
+            return trigger;
+        }
+
+        private async Task<long> GetEstimatedWork (CosmosDBTrigger trigger)
+        {
+            try
+            {
+                var builder = CreateChangeFeedProcessorBuilder(trigger);
+                var workEstimator = await builder.BuildEstimatorAsync();
+                long estimatedRemainingWork = await workEstimator.GetEstimatedRemainingWork();
+                return estimatedRemainingWork;
+            }
+            catch(Exception ex)
+            {
+                _logger.LogError(ex, "Error getting estimated work from host");
+                throw ex;
+            }
+        }
+
+        private static ChangeFeedProcessorBuilder CreateChangeFeedProcessorBuilder(
+            CosmosDBTrigger trigger)
+        {
+            CosmosDBConnectionString triggerConnection = new CosmosDBConnectionString(trigger.CosmosDBConnectionString);
+            DocumentCollectionInfo documentCollectionLocation = new DocumentCollectionInfo
+            {
+                Uri = triggerConnection.ServiceEndpoint,
+                MasterKey = triggerConnection.AuthKey,
+                DatabaseName = trigger.DatabaseName,
+                CollectionName = trigger.CollectionName
+            };
+
+            CosmosDBLease lease = trigger.Lease;
+            CosmosDBConnectionString leaseConnection = new CosmosDBConnectionString(lease.LeasesCosmosDBConnectionString);
+
+            DocumentCollectionInfo leaseCollectionLocation = new DocumentCollectionInfo
+            {
+                Uri = leaseConnection.ServiceEndpoint,
+                MasterKey = leaseConnection.AuthKey,
+                DatabaseName = lease.LeaseDatabaseName,
+                CollectionName = lease.LeaseCollectionName
+            };
+
+            HostPropertiesCollection hostProperties = new HostPropertiesCollection(documentCollectionLocation, leaseCollectionLocation);
+
+            var changeFeedClient = new DocumentClient(documentCollectionLocation.Uri, documentCollectionLocation.MasterKey);
+            IChangeFeedDocumentClient feedDocumentClient = new ChangeFeedDocumentClient(changeFeedClient);
+
+            var leaseClient = new DocumentClient(leaseCollectionLocation.Uri, leaseCollectionLocation.MasterKey);
+            IChangeFeedDocumentClient leaseDocumentClient = new ChangeFeedDocumentClient(leaseClient);
+
+            var builder = new ChangeFeedProcessorBuilder()
+                            .WithHostName(hostProperties.HostName)
+                            .WithFeedCollection(documentCollectionLocation)
+                            .WithLeaseCollection(leaseCollectionLocation)
+                            .WithFeedDocumentClient(feedDocumentClient)
+                            .WithLeaseDocumentClient(leaseDocumentClient);
+            return builder;
+        }
+
+        private static string NormalizeString(string inputString)

```

Comment:
move this to `Utilities` class and add unit tests

Reply:
Let's move it to `Extensions/StringExtensions` instead, otherwise Utilities will become a dump :D